File: leetcode/exercises/1.2.1.easy.two_sum_1.js
// -----------------------
// Created     : 24/10/2025
// Last Edited : 24/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 1
// Source      :
// Notes       :
// -----------------------

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {};
    for (const [i, value] of nums.entries())
    {
        const complement = target - value;
        if (complement in map) { return [map[complement], i] };
        map[value] = i;
    };
};

console.log(twoSum([1, 2, 3, 10, 4, 8, 7], 12))

File: leetcode/exercises/1.2.1.easy.two_sum_1.go
// -----------------------
// Created     : 30/10/2025
// Last Edited : 30/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 1
// Source      :
// Notes       :
// -----------------------

package main
import "fmt"

func twoSum(nums []int, target int) []int {
    m := make(map[int]int{})
    for i, v := range nums {
        complement := target - v
        _, ok := m[complement] 
        if ok { return []int{m[complement], i} }
        m[v] = i
    }
    return []int{0,0}
}

func main() {
    fmt.Println(twoSum([]int{1,5,6,2,2,8}, 11))
}
